jobs/kpi-forecasting/kpi_forecasting/models/prophet_forecast.py

```diff
@@ -3,65 +3,339 @@
+            Parameter modulating the strength of the holiday
+            components model, unless overridden in the holidays input.
+        changepoint_prior_scale: Used in Prophet object initialization
+            Parameter modulating the flexibility of the
+            automatic changepoint selection. Large values will allow many
+            changepoints, small values will allow few changepoints.
+        mcmc_samples (int): Used in Prophet object initialization
+            If greater than 0, will do full Bayesian inference
+            with the specified number of MCMC samples. If 0, will do MAP
+            estimation.
+        interval_width (float): Used in Prophet object initialization
+            width of the uncertainty intervals provided
+            for the forecast. If mcmc_samples=0, this will be only the uncertainty
+            in the trend using the MAP estimate of the extrapolated generative
+            model. If mcmc.samples>0, this will be integrated over all model
+            parameters, which will include uncertainty in seasonality.
+        uncertainty_samples: Used in Prophet object initialization
+            Number of simulated draws used to estimate
+            uncertainty intervals. Settings this value to 0 or False will disable
+            uncertainty estimation and speed up the calculation.
+        stan_backend (str): Used in Prophet object initialization
+             str as defined in StanBackendEnum default: None - will try to
+            iterate over all available backends and find the working one
+        holidays_mode (str): Used in Prophet object initialization
+            'additive' or 'multiplicative'. Defaults to seasonality_mode.
     """
 
-    number_of_simulations: int = 1000
+    holidays: list = field(default_factory=list[ProphetHoliday])
+    regressors: list = field(default_factory=list[ProphetRegressor])
+    use_all_us_holidays: bool = False
+
+    # these are the arguments used to initialize the Prophet object
+    growth: str = "linear"
+    changepoints: list = None
+    n_changepoints: int = 25
+    changepoint_range: float = 0.8
+    yearly_seasonality: str = "auto"
+    weekly_seasonality: str = "auto"
+    daily_seasonality: str = "auto"
+    holidays: pd.DataFrame = None
+    seasonality_mode: str = "additive"
+    seasonality_prior_scale: float = 10.0
+    holidays_prior_scale: float = 10.0
+    changepoint_prior_scale: float = 0.05
+    mcmc_samples: int = 0
+    interval_width: float = 0.80
+    uncertainty_samples: int = 1000
+    stan_backend: str = None
+    scaling: str = "absmax"
+    holidays_mode: str = None
+    floor: float = None
+    cap: float = None
+
+    def __post_init__(self):
+        holiday_list = []
+        regressor_list = []
+
+        if self.holidays == []:
+            self.holidays = None
+            self.holidays_raw = None
+        elif not self.holidays:
+            self.holidays_raw = None
```

Comment:
What are these different conditionals checking for? The first one is clearly checking for an empty list. Is the second one checking if self.holidays is None?

Reply:
The idea was that its possible to have an empty list for holidays in the config, and in that case things would get screwed up in the `elif` case.  So this ensures that an empty list is treated the same as just leaving it out of the config.

The whole `holidays_raw` and `holidays` thing is annoying.  It comes down to the fact we have an intermediate class to parse the holidays in the config into the dataframe (oh prophet...) that is expected.  And if I want to easily create new prophet classes it's easier to distinguish between the "raw" non-df holiday format and the 'not raw' df format that can be passed to prophet under the `holidays` keyword argument.  It's annoying but I couldn't think of a cleaner way.